### ISSUE TITLE
Completion gets worse after turning on R console

### ISSUE THREAD

**Issue by dinga92** [R-nvim/R.nvim#95]
Hi, I am not sure if this is an expected behavior, or a bug, or a wrong config on my side. 

I am observing behavior when I open an .R file the completion works according to what is in the file, so for example, if file contains `library(ggplot2)` I will be offered all ggplot variables for the completion, but then I run R using `<leader>rf` completion no longer works for ggplot, it works only for base R things and it works for ggplot again only after I run the code in the R console. Sorry for a convoluted explanation, I hope it is understandable anyway.

**Comment by jalvesaq** [comment]
Yes, this is the expected behavior. We can discuss and change the expected behavior, but, currently, both the Object Browser and auto-completion reflect what is loaded in R's memory. Suppose that you are sending lines one at a time and that a specific `library()` call is in the middle of the script. It would not work if you tried to call a function of the unloaded library before sending its `library()` command to R. Consistent with this, auto-completion will not work before the library is loaded. In the rare circumstance when a user detaches a library, it's also unloaded from `rnvimserver`.

But, as I said, we can change the default behavior to:

1. Load in `rnvimserver` the data of all libraries with a corresponding `library()` command in the script.
2. After starting R, never unload any library data from `rnvimserver`.
3. Load in `rnvimserver` data of libraries that were not in the script at R startup as they are loaded after R was started.

**Comment by dinga92** [comment]
alright, got it

**Comment by PMassicotte** [comment]
Is this still relevant given we are now using `r_ls`?

**Comment by jalvesaq** [comment]
The "issue" didn't change with the LSP.

It would be possible to create an option to allow users to choose between two behaviors after R is started:

- Complete only objects of loaded packages (current behavior, based on actual R's memory).
- Complete objects of packages in `library` and `require` commands in the current script (based on the R script being edited).

However, the LSP code is already complex, and adding another layer would make it harder to maintain. In my opinion, it's better to choose one of the two behaviors. I prefer the current behavior. If nobody wants to implement the option to alternate between the two behaviors, we can either keep this "issue" open forever or close it as a feature request that won't be implemented.

**Comment by jalvesaq** [comment]
Maybe we could close this...

**Comment by PMassicotte** [comment]
I agree.
